linkerd/app/outbound/src/tcp/connect.rs

```diff
@@ -25,19 +33,24 @@ impl Outbound<()> {
 }
 
 impl<C> Outbound<C> {
-    pub fn push_tcp_endpoint<P>(
+    pub fn push_tcp_endpoint<T>(
         self,
     ) -> Outbound<
         impl svc::Service<
-                Endpoint<P>,
+                T,
                 Response = impl io::AsyncRead + io::AsyncWrite + Send + Unpin,
                 Error = Error,
                 Future = impl Send,
             > + Clone,
     >
     where
-        Endpoint<P>: svc::Param<Option<SessionProtocol>>,
-        C: svc::Service<Endpoint<P>, Error = io::Error> + Clone + Send + 'static,
+        T: svc::Param<Remote<ServerAddr>>
+            + svc::Param<tls::ConditionalClientTls>
+            + svc::Param<Option<opaque_transport::PortOverride>>
+            + svc::Param<Option<http::AuthorityOverride>>
+            + svc::Param<Option<SessionProtocol>>
+            + svc::Param<transport::labels::Key>,
```

Comment:
whoa, hmm. part of me wonders whether it might be worth making a trait alias for "all these params", but that might just be unnecessary complexity...

Reply:
Would a trait alias actually reduce the complexity? We generally don't have to copy them everywhere. So the trait alias would be at least as much code...

I think this probably good/fine as-is--for now, at least. I think we want each of these stack modules to basically take a generic target that provides a set of params; and then it builds its inner target type as necessary (either wrapping or replaces the initial target). Once we've wired this approach we may find a way to simplify it further.

Reply:
yeah, we're on the same page. i was just thinking about it, but i think this is actually better than wrapping it into one trait.